### DISCUSSION TITLE
Ubuntu 20.04 support notes

### DISCUSSION THREAD

**Discussion by birdybro** [retronas/discussions#107] [Ideas]
Just some notes while I try to get this going on Ubuntu 20.04 for an interesting non-retro gaming purpose of my own (legacy hardware that needs old smb support in a mixed environment with new hardware on a secure segregated off the grid network, it's complicated :P, using an easy solution like this would be nice).

1. Due to no default user named pi, the user setup could probably benefit from autogenerating a user upon typing in a name that is not found. This user's group membership, wheel, etc... all could be checked with fact gathering, and configuration automated.
2. OpenSSH-server fails install on ubuntu 20.04 because "restart ssh" is not valid for Ubuntu for whatever reason. I'm not sure what the script tries to do here specifically, it should be doing `systemctl restart ssh` as it's already running in an elevated prompt. One relevant line --> https://github.com/danmons/retronas/blob/20d4b587058553816420b903d0727ec8a7a05067/ansible/install_openssh.yml#L19 - When I exited the program and manually restarted the ssh service, I was able to rerun the install script for openssh-server, so this seems to be running fine after one little hiccup.
3. For cockpit installation the debian signing keys 1 step fails (expectedly). A detection of ubuntu and then appropriate handling in light of the different OS probably would have to be developed.
![image](https://user-images.githubusercontent.com/16388068/152017360-47e6b10e-25fd-4636-a578-bbad70e33f5a.png)
4. setting up the samba share must have skipped over some permissions, looking into that and all the configuration looks correct, but I assume the version of samba must be different in 20.04 vs the latest RPi OS 64-bit, hard to explain otherwise off the top of my head.

That's all I got so far. Hopefully this helps as a reference point.

**Comment by birdybro** [comment]
Side Note:

Debian 11 works just fine in a Hyper-V VM, worked very well for anyone curious.

**Comment by sairuk** [comment]
I've recently added a check that the user exists that leans on getent as part of the username change dialog, I've also added similar logic for the EtherDFS interface, they are part of a larger update I am looking into to break out the menu data and dialog generation.

I am going to add similar logic to the startup script to prompt the user to change to an existing user if the current one does not exist for the deployed system as you demonstrate above the reliance on a `pi` user without checks and balances is breaking, i've been caught out a number of times in testing

The [dialog-rewrite](https://github.com/danmons/retronas/tree/dialog-rewrite) branch includes this new logic but also is a large step in the code base which I'd prefer was well tested before I merge to main
